lib/console/gateways/gateways.ex

```diff
@@ -90,7 +95,8 @@ defmodule Console.Gateways do
 
   """
   def delete_gateway(%Gateway{} = gateway) do
-    Repo.delete(gateway)
+    Repo.get!(GatewayIdentifier, gateway.gateway_identifier_id)
```

Comment:
if you add `on_delete: :delete_all` to the relationship on the gateway side, you should be able to just  delete the gateway and that will cascade to the hardware identifiers